### PR TITLE
Update get mentor categories endpoint (public) to send category IDs

### LINKED ISSUE #82: Update get mentor categories endpoint (public) to send category IDs
**Description:**
Currently, the `api/categories` endpoint only returns a list of category names. The category IDs are required during the mentor application process. 

**Tasks:**
Add the category IDs to the response.

**Acceptance Criteria:**
Should be able to get all categories with their IDs.
- Sample response: 
```json
{
    "categories": [
        {
            "category": "Data Science",
            "id": "95ba00c0-8173-43c4-9573-6b481b4085e2"
        },
        {
            "category": "Random Category",
            "id": "fe7ca629-4f5e-440a-86cb-d54f8b98ec5f"
        },
        {
            "category": "Computer Science",
            "id": "fef68adb-e710-4d9e-8772-dc4905885088"
        }
    ],
    "message": "All Categories found"
}
```

**Additional Information:**
Endpoint implemented with #16

**Related Dependencies or References:**

### PR DESCRIPTION
**Pull request by Madhawa97**
## Purpose

The purpose of this PR is to fix #82 

## Goals
Currently, the api/categories endpoint only returns a list of category names. The category IDs are also required during the mentor application process.

## Approach
Add the category IDs to the response.
Sample response: 
```json
{
    "categories": [
        {
            "category": "Data Science",
            "id": "95ba00c0-8173-43c4-9573-6b481b4085e2"
        },
        {
            "category": "Random Category",
            "id": "fe7ca629-4f5e-440a-86cb-d54f8b98ec5f"
        },
        {
            "category": "Computer Science",
            "id": "fef68adb-e710-4d9e-8772-dc4905885088"
        }
    ],
    "message": "All Categories found"
}
```

### Screenshots
N/A
##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by Madhawa97** [comment]
Tests are passing locally.
image

File: src/services/category.service.ts [hunk 1]
```diff
@@ -1,18 +1,25 @@
 import { dataSource } from '../configs/dbConfig'
 import Category from '../entities/category.entity'
 
+interface CategoryType {
+  category: string
+  id: string
+}
```
Comment (anjula-sack) [inline src/services/category.service.ts:7]:
```suggestion
interface Category {
  category: string
  id: string
}
```

File: src/services/category.service.ts [hunk 2]
```diff
@@ -1,18 +1,25 @@
...
+
 export const getAllCategories = async (): Promise<{
   statusCode: number
-  categories?: string[] | null
+  categories?: CategoryType[] | null
```
Comment (anjula-sack) [inline src/services/category.service.ts:11]:
```suggestion
  categories?: Category[] | null
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Comment by anjula-sack** [comment]
@Madhawa97 Can you add the requested changes?

**Comment by Madhawa97** [comment]
Hi @anjula-sack, Can you check now, please?

**Review by anjula-sack — APPROVED** [review]
Thank you @Madhawa97

**Pull request merged**